lighthouse-core/lib/page-functions.js

```diff
@@ -138,7 +138,17 @@ function getOuterHTMLSnippet(element, ignoreAttrs = [], snippetCharacterLimit =
         dirty = true;
       }
 
-      if (dirty) clone.setAttribute(attributeName, attributeValue);
+      if (dirty) {
+        // Style attributes can be blocked by the CSP if they are set via `setAttribute`.
```

Comment:
love finding those CSP holes :D

Reply:
IIRC constructable stylesheets are also not constrained by CSP but allow you to do the same stuff that is.  🤷